server/pkg/tunnel/config/config.go

```diff
@@ -9,12 +17,41 @@ type TunnelServerConfig struct {
 	// default true
 	Enable bool `json:"enable,omitempty"`
 	// TunnelACLConfig indicates the set of tunnel server config about acl
-	acl.TunnelACLConfig
+	TunnelACLConfig acl.TunnelACLConfig `json:"tunnelACL,omitempty"`
 	// NodeName indicates the node name of tunnel server
 	NodeName string `json:"nodeName,omitempty"`
 	// ListenPort indicates the listen port of tunnel server
 	// default 20004
 	ListenPort int `json:"listenPort,omitempty"`
 	// PublicIP indicates the public ip of tunnel server
 	PublicIP string `json:"publicIP,omitempty"`
+	// EnableSecurity indicates whether to use the ca acl and security transport
+	// default false
+	EnableSecurity bool `json:"enableSecurity"`
+}
+
+func NewTunnelServerConfig() *TunnelServerConfig {
+	nodeName, isExist := os.LookupEnv(meshConstants.MY_NODE_NAME)
+	if !isExist {
+		klog.Fatalf("env %s not exist", meshConstants.MY_NODE_NAME)
+	}
+
+	publicIP := util.FetchPublicIP()
+	if publicIP == "" {
+		publicIP = "0.0.0.0"
+	}
+	klog.Infof("Fetch public IP: %s", publicIP)
+
+	return &TunnelServerConfig{
+		Enable: true,
+		TunnelACLConfig: acl.TunnelACLConfig{
+			TLSPrivateKeyFile: meshConstants.ServerDefaultKeyFile,
+			TLSCAFile:         meshConstants.ServerDefaultCAFile,
+			TLSCertFile:       meshConstants.ServerDefaultCertFile,
+		},
+		NodeName:       nodeName,
+		ListenPort:     20004,
```

Comment:
const variable for here listerPort 20004

Reply:
fix